packages/compiler/core/checker.ts

```diff
@@ -4658,26 +4671,54 @@ export function filterModelProperties(
   return finishTypeForProgram(program, newModel);
 }
 
+/**
+ * Gets the property from the nearest base type that is overridden by the
+ * given property, if any.
+ */
+export function getOverriddenProperty(property: ModelProperty): ModelProperty | undefined {
+  compilerAssert(
+    property.model,
+    "Parent model must be set before overridden property can be found."
+  );
+
+  for (let current = property.model.baseModel; current; current = current.baseModel) {
+    const overriden = current.properties.get(property.name);
+    if (overriden) {
+      return overriden;
+    }
+  }
```

Comment:
```suggestion
  for (let current = property.model.baseModel; current; current = current.baseModel) {
    const overridden = current.properties.get(property.name);
    if (overridden) {
      return overridden;
    }
  }
```

Looks like I left you a spell check failure, oops.